modules/mysql.go

```diff
@@ -0,0 +1,129 @@
+package modules
+
+import (
+	"net"
+
+	logrus "github.com/sirupsen/logrus"
+	"github.com/zmap/zgrab2"
+	"github.com/zmap/zgrab2/lib/mysql"
+)
+
+var logger *logrus.Logger
+
+// HandshakeLog contains detailed information about each step of the
+// MySQL handshake, and can be encoded to JSON.
+type MySQLScanResults struct {
+	mysql.ConnectionLog
+	TLSLog *zgrab2.TLSLog `json:"tls,omitempty"`
+}
+
+type MySQLFlags struct {
+	zgrab2.BaseFlags
+	zgrab2.TLSFlags
+	Verbose bool `long:"verbose" description:"More verbose logging, include debug fields in the scan results"`
+}
+
+type MySQLModule struct {
+}
+
+type MySQLScanner struct {
+	config *MySQLFlags
+}
+
+func init() {
+	logger = logrus.New()
+	var module MySQLModule
+	_, err := zgrab2.AddCommand("mysql", "MySQL", "Grab a MySQL handshake", 3306, &module)
+	if err != nil {
+		logger.Fatal(err)
+	}
+}
+
+func (m *MySQLModule) NewFlags() interface{} {
+	return new(MySQLFlags)
+}
+
+func (m *MySQLModule) NewScanner() zgrab2.Scanner {
+	return new(MySQLScanner)
+}
+
+func (f *MySQLFlags) Validate(args []string) error {
+	return nil
+}
+
+func (f *MySQLFlags) Help() string {
+	return ""
+}
+
+func (s *MySQLScanner) Init(flags zgrab2.ScanFlags) error {
+	f, _ := flags.(*MySQLFlags)
+	s.config = f
+	if f.Verbose {
+		logger.SetLevel(logrus.DebugLevel)
+	}
+	return nil
+}
+
+func (s *MySQLScanner) InitPerSender(senderID int) error {
+	return nil
+}
+
+func (s *MySQLScanner) GetName() string {
+	return s.config.Name
+}
+
+func (s *MySQLScanner) GetPort() uint {
+	return s.config.Port
+}
+
+func (s *MySQLScanner) Scan(t zgrab2.ScanTarget) (status zgrab2.ScanStatus, _result interface{}, thrown error) {
+	sql := mysql.NewConnection(&mysql.Config{
+		Host: t.IP.String(),
+		Port: uint16(s.config.Port),
+	})
+
+	result := MySQLScanResults{}
+	_result = &result
+	defer func() {
```

Comment:
It seems like the pre-panic implementation was tangle of ifs, and this seemed the natural solution.

But, that could very well be because I'm used to thinking in try/catch blocks.

 - [ ] Try again to find a sensible non-panic()ing implementation